C/cp/max_sum_subarray.c

```diff
@@ -0,0 +1,59 @@
+#include<stdio.h>
+
+int max(int a,int b) {
+    return (a >= b) ? a : b;
+}
+
+int max_sum_crossing(int arr[], int lo, int mid, int hi) {
+    int cur = 0, presum = -1e9;
+    for (int i = mid; i >= lo; --i) {
+        cur += arr[i];
+        presum = max(presum, cur);
+    }
+    cur = 0;
+    int postsum = -1e9;
+    for (int i = mid + 1; i <= hi; ++i) {
+        cur += arr[i];
+        postsum = max(postsum, cur);
+    }
+
+    return max(max(presum, postsum), presum + postsum);
+
+}
+
+int max_sum_subarray(int arr[], int lo, int hi) {
+    if(lo == hi) {
+        return arr[lo];
+    }
+    int mid = (lo + hi) / 2;
```

Comment:
@SangeetaMishra Use `mid = lo + ((hi - lo) / 2)` to avoid integer overflow condition.